datastore/data-loader/client_knmi_station_ingest.py

```diff
@@ -0,0 +1,254 @@
+    station_name = station_name.lower()
+    if level_raw := re.search(regex_level, long_name):
+        level = level_raw[0]
+    if level_raw := re.search(regex_level_centimeters, long_name):
+        level = str(float(level_raw[0]) / 100.0)
+    elif "grass" in long_name:
+        level = "0"
+    elif param_id in ["pg", "pr", "pwc", "vv", "W10", "W10-10", "ww", "ww-10", "za", "zm"]:
+        # https://english.knmidata.nl/open-data/actuele10mindataknmistations
+        # Comments code: 2, 3, 11
+        # Note: The sensor is not installed at equal heights at all types of measurement sites:
+        # At 'AWS' sites the device is installed at 1.80m. At 'AWS/Aerodrome' and 'Mistpost'
+        # (note that this includes site Voorschoten (06215) which is 'AWS/Mistpost')
+        # the device is installed at 2.50m elevation. Exceptions are Berkhout AWS (06249),
+        # De Bilt AWS (06260) and Twenthe AWS (06290) where the sensor is installed at 2.50m.
+        # Since WaWa is automatic detection I asssumed that the others stations are AWS, thus 1.80m
+        if (
+            station_id in ["06215", "06249", "06260", "06290"]
+            or "aerodrome" in station_name
+            or "mistpost" in station_name
+        ):
+            level = "2.50"
+        else:
+            level = "1.80"
+
+    # We want "level" to be numeric, so get rid of "first", "second" and "third".
+    # Note that this level has no meaning, it is just a hack for this test dataset
+    if level == "first":
+        level = "0"
+    if level == "second":
+        level = "1"
+    if level == "third":
+        level = "2.0"
+
+    if "minimum" in long_name:
+        function = "minimum"
+    elif "maximum" in long_name:
+        function = "maximum"
+    elif "average" in long_name:
+        function = "mean"
+    else:
+        function = "point"
+
+    period = "PT0S"
+    if period_raw := re.findall(regex_time_period, long_name):
+        if len(period_raw) == 1:
+            period_raw = period_raw[0]
+        else:
+            raise Exception(f"{period_raw}, {long_name}")
+        time, scale = period_raw
+        if scale == "hours":
+            period = f"PT{time}H"
+        elif scale == "min":
+            period = f"PT{time}M"
+    elif param_id == "ww-10":
+        period = "PT10M"
+    elif param_id == "ww":
+        period = "PT01H"
+
+    standard_name = convert_standard_names_to_cf(standard_name)
+    return standard_name, level, function, period
+
+
+def send_request_to_ingest(msg, url):
+    try:
+        response = requests.post(url, data=json.dumps(msg))
+        response.raise_for_status()
+        return response.status_code, response.json()
+    except requests.RequestException as e:
+        return response.status_code, e
+
+
+def insert_data(observation_request_messages: List, url):
+    with Pool(cpu_count()) as pool:
+        results = pool.starmap(send_request_to_ingest, [(msg, url) for msg in observation_request_messages])
```

Comment:
Nice trick, didn't know about the `starchart` function.
I would have gone for `functools.partial` but this works as well. And it is good to know it exists. Although this is a slight abuse, I would say, as the `url` is fixed.

Reply:
They have the same performance and it does the same as `functools.partial` in one line. `Starmap` is flexible, but maybe `partial` could be a bit more readable and concise, since the `url` stays the same. 

So refactored to use `functools.partial` in https://github.com/EURODEO/e-soh/pull/209/commits/603ffef4e63f0e5b53b0061d7d7f3ba7f7aa5c41